ios/PurchasesHybridCommon/PurchasesHybridCommonTests/PurchasesHybridCommonTests.swift

```diff
@@ -197,5 +197,221 @@ class PurchasesHybridCommonTests: QuickSpec {
             }
         }
 
+        context("beginRefundRequest") {
+            if #available(iOS 15.0, *) {
+                context("productId") {
+                    it("passes the call correctly to Purchases") {
+                        let mockPurchases = MockPurchases()
+
+                        CommonFunctionality.sharedInstance = mockPurchases
+
+                        CommonFunctionality.beginRefundRequest(productId: "mock-product-id") { _ in }
+
+                        expect(mockPurchases.invokedBeginRefundRequestForProductCount) == 1
+                        expect(mockPurchases.invokedBeginRefundRequestForProductParameters?
+                            .productId) == "mock-product-id"
+                    }
+
+                    it("does not return an error if successful") {
+                        let mockPurchases = MockPurchases()
+
+                        CommonFunctionality.sharedInstance = mockPurchases
+
+                        var completionCallCount = 0
+                        var completionError: ErrorContainer? = nil
+
+                        CommonFunctionality.beginRefundRequest(productId: "mock-product-id") { error in
+                            completionCallCount += 1
+                            completionError = error
+                        }
+                        mockPurchases.invokedBeginRefundRequestForProductParameters?.1(Result(.success, nil))
+
+                        expect(completionCallCount) == 1
+                        expect(completionError).to(beNil())
+                    }
+
+                    it("returns an error with userCancelled extra info set to true") {
+                        let mockPurchases = MockPurchases()
+
+                        CommonFunctionality.sharedInstance = mockPurchases
+
+                        var completionCallCount = 0
+                        var completionError: ErrorContainer? = nil
+
+                        CommonFunctionality.beginRefundRequest(productId: "mock-product-id") { error in
+                            completionCallCount += 1
+                            completionError = error
+                        }
+                        mockPurchases.invokedBeginRefundRequestForProductParameters?.1(Result(.userCancelled, nil))
```

Comment:
This is equivalent to
```suggestion
                        mockPurchases.invokedBeginRefundRequestForProductParameters?.1(.success(.userCancelled))
```

I wouldn't use the `Result(value, error)` overload just to set it to a success.